HARK/distribution.py

```diff
@@ -930,29 +929,288 @@ def draw(self, N, X=None, exact_match=False):
+    def pmf(self):
+        """
+        Returns the distribution's probability mass function.
+        """
+        return self._xarray.pmf
+
+    @property
+    def RNG(self):
+        """
+        Returns the distribution's random number generator.
+        """
+        return self._xarray.RNG
+
+    @property
+    def data(self):
+        """
+        The distribution's data as an array. The underlying
+        array type (e.g. dask, sparse, pint) is preserved.
+        """
+        return self._xarray.data
+
+    @property
+    def coords(self):
+        """
+        The distribution's coordinates.
+        """
+        return self._xarray.coords
+
+    @property
+    def dims(self):
+        """
+        The distribution's dimensions.
+        """
+        return self._xarray.dims
+
+    @property
+    def name(self):
+        """
+        The distribution's name.
+        """
+        return self._xarray.name
+
+    @property
+    def attrs(self):
+        """
+        The distribution's attributes.
+        """
+        return self._xarray.attrs
+
+    def expected_value(self, func=None, *args, labels=False):
+        """
+        Expectation of a function, given an array of configurations of its inputs
+        along with a DiscreteDistributionXRA object that specifies the probability
+        of each configuration.
+
+        Parameters
+        ----------
+        func : function
+            The function to be evaluated.
+            This function should take the full array of distribution values
+            and return either arrays of arbitrary shape or scalars.
+            It may also take other arguments *args.
+            This function differs from the standalone `calc_expectation`
+            method in that it uses numpy's vectorization and broadcasting
+            rules to avoid costly iteration.
+            Note: If you need to use a function that acts on single outcomes
+            of the distribution, consier `distribution.calc_expectation`.
+        *args :
+            Other inputs for func, representing the non-stochastic arguments.
+            The the expectation is computed at f(dstn, *args).
+        labels : bool
+            If True, the function should use labeled indexing instead of integer
+            indexing using the distribution's underlying rv coordinates. For example,
+            if `dims = ('rv', 'x')` and `coords = {'rv': ['a', 'b'], }`, then
+            the function can be `lambda x: x["a"] + x["b"]`.
+
+        Returns
+        -------
+        f_exp : np.array or scalar
+            The expectation of the function at the queried values.
+            Scalar if only one value.
+        """
+
+        def func_wrapper(x, *args):
+            """
+            Wrapper function for `func` that handles labeled indexing.
+            """
+            dim_0 = self.dims[0]
+            idx = self.coords[dim_0].values
+
+            wrapped = dict(zip(idx, x))
+
+            return func(wrapped, *args)
```

Comment:
I would like to talk to you about what you are doing with the 'labels' case here before merging.
You asked if there is a simpler way to do what you're trying to do, and I suspect there is.
But the documentation doesn't make it clear to me what's happening.